app/views/layouts/_navbar.html.haml

```diff
@@ -1,3 +1,8 @@
-= link_to "Create Article", new_article_path
-= link_to "Sign In", new_user_session_path
-= link_to "Sign Up", new_user_registration_path
+- if user_signed_in?
+  = "Signed in as: #{current_user.role.capitalize} (#{current_user.email}) |"
+  = link_to("Create Article", new_article_path) + "|" if policy(Article).new?
```

Comment:
Please be consistent with the usage of parathesis

Reply:
We skip the parenthesis when possible to take advantage of the beauty of ruby.
But in this specific case, they are required so the pipe " | " gets involved in the conditional statement.